apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -225,173 +134,85 @@ export class TeacherFeedback extends Component {
 
     return (
       <div style={tabVisible}>
-        {this.state.errorState === ErrorType.Load && (
-          <span>
-            <i className="fa fa-warning" style={styles.errorIcon} />
-            {i18n.feedbackLoadError()}
-          </span>
-        )}
-        {this.props.rubric && (
-          <div style={styles.performanceArea}>
-            <div style={styles.keyConceptArea}>
-              <h1 style={styles.h1}> {i18n.rubricKeyConceptHeader()} </h1>
-              <p style={styles.keyConcepts}>{this.props.rubric.keyConcept}</p>
-            </div>
-            <div style={styles.rubricArea}>
-              <h1 style={styles.h1}> {i18n.rubricHeader()} </h1>
-              <form style={styles.form}>
-                {rubricLevels.map(level => (
-                  <RubricField
-                    key={level}
-                    showFeedbackInputAreas={showFeedbackInputAreas}
-                    expandByDefault={
-                      this.props.displayKeyConcept ||
-                      (expandPerformanceLevelForStudent &&
-                        this.state.performance === level)
-                    }
-                    rubricLevel={level}
-                    rubricValue={this.props.rubric[level]}
-                    disabledMode={this.props.disabledMode}
-                    onChange={this.onRubricChange}
-                    currentlyChecked={this.state.performance === level}
-                  />
-                ))}
-              </form>
-            </div>
-          </div>
+        {errorState === ErrorType.Load &&
+          this.renderError(i18n.feedbackLoadError())}
+        {rubric && (
+          <TeacherFeedbackRubric
+            rubric={rubric}
+            performance={performance}
+            displayKeyConcept={displayKeyConcept}
+            disabledMode={disabledMode}
+            onRubricChange={this.onRubricChange}
+            viewAs={viewAs}
+          />
         )}
-        {!this.props.displayKeyConcept && !dontShowStudentComment && (
+        {!displayKeyConcept && !dontShowStudentComment && (
           <div style={styles.commentAndFooter}>
             <CommentArea
-              disabledMode={this.props.disabledMode}
-              comment={this.state.comment}
+              disabledMode={disabledMode}
+              comment={comment}
               placeholderText={placeholderText}
               studentHasFeedback={
-                this.props.viewAs === ViewType.Student &&
-                this.state.latestFeedback.length > 0
+                viewAs === ViewType.Student && !!latestFeedback
               }
               onCommentChange={this.onCommentChange}
             />
             <div style={styles.footer}>
-              {this.props.viewAs === ViewType.Teacher && (
+              {viewAs === ViewType.Teacher && (
                 <div style={styles.button}>
                   <Button
                     __useDeprecatedTag
```

Comment:
When you make functional changes in a future PR, it would be nice to get rid of __useDeprecatedTag.

Reply:
Will do. Do you know the background on why the tag was added?

Reply:
This is what I've heard: The old version of the Button component used <a> or <div>.  That was deprecated at some point in favor of <button> but there was some reason that we couldn't just move to the new implementation wholesale.  Buttons that needed to keep the old behavior used the deprecated tag (which would hopefully get cleaned up over time).  I believe Jessie has more context if you need more details.